### ISSUE TITLE
Duplication of orgs in sharing groups

### ISSUE THREAD

**Issue by zpevma** [MISP/MISP#7867]
### Work environment

I have quite simple setup:
 - user admin@admin.test who is member of organization: **ORGNAME** and has **admin** role
 - sharing group **TestingSG** with one member organization ORGNAME
 - event **TEST EVENT SG** where Creator org=Owner org=ORGNAME, distribution set to sharing group TestingSG

Simple script which updates the event:
```
from pymisp import MISPEvent, ExpandedPyMISP

misp = ExpandedPyMISP('URL', 'KEY',  ssl=False, debug=False)

srch_rsp = misp.search_index(eventinfo='TEST EVENT SG')

event = MISPEvent()
event.load({'Event': srch_rsp[0]})
event.timestamp = None
misp.update_event(event_id=event.id, event=event)
```

The issue is that after each run of this script, the organization ORGNAME is added to sharing group again, for example, after running the script 3 times, the sharing group looks like this:
![image](https://user-images.githubusercontent.com/14806666/138113808-999c6b51-4cf8-47a7-9ddf-91ffe5beeae9.png)

| Questions                 | Answers
|---------------------------|--------------------
| Type of issue             | Bug
| OS version (server)       | Ubuntu 20.04
| OS version (client)       | Ubuntu 20.04
| PHP version               | 7.4.3
| MISP version / git hash   | 2.4.148
| Browser                   |   Firefox/93.0

### Expected behavior
Don't add the organization to SG again, just update the event.

### Actual behavior
Organization is added to sharing group.

**Comment by JakubOnderka** [comment]
Should be fixed in #7871.

**Comment by zpevma** [comment]
Thanks @JakubOnderka for such a quick response!

**Comment by github-germ** [comment]
@JakubOnderka ... Happy holidays to you...

Aha, we just noticed 1000s of dup rows in the `sharing_groups_org` table (which was causing HUGE results (>10GB) for some `pymisp.search()` results!

Glad to see this was fixed in 2.4.151. **THANKS**.

A few questions so we can clean this in our instances:

1. Does the schema change adding the `name` column as an index in the `sharing_groups` table get patched when 2.4.151 is installed?
2. Should we delete all the dups via mysql or does something mitigate that when 2.4.151 is installed?
3. Anything else?
